internal/cimfs/common.go

```diff
@@ -0,0 +1,109 @@
+package cimfs
+
+import (
+	"syscall"
+	"time"
+)
+
+var (
+	// Equivalent to SDDL of "D:NO_ACCESS_CONTROL"
+	nullSd = []byte{1, 0, 4, 128, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
+)
+
+// 100ns units between Windows NT epoch (Jan 1 1601) and Unix epoch (Jan 1 1970)
+const epochDelta = 116444736000000000
+
+// Filetime is a Windows FILETIME, in 100-ns units since January 1, 1601.
+type Filetime int64
+
+// Time returns a Go time equivalent to `ft`.
+func (ft Filetime) Time() time.Time {
+	if ft == 0 {
+		return time.Time{}
+	}
+	return time.Unix(0, (int64(ft)-epochDelta)*100)
+}
+
+func FiletimeFromTime(t time.Time) Filetime {
+	if t.IsZero() {
+		return 0
+	}
+	return Filetime(t.UnixNano()/100 + epochDelta)
+}
+
+func (ft Filetime) String() string {
+	return ft.Time().String()
+}
+
+func (ft Filetime) toSyscallFiletime() syscall.Filetime {
+	return syscall.NsecToFiletime((int64(ft) - epochDelta) * 100)
+}
+
+// A FileInfo specifies information about a file.
+type FileInfo struct {
```

Comment:
This is basically the same as the definition in `winapi` can we use that?

Reply:
I tried really hard to do that but the CIMFS APIs expect a struct definition that is not exactly same as this and the functions which use this definition can't be easily modified to use the other struct definition.